### ISSUE TITLE
Cannot provision nodeport in services because of limit

### ISSUE THREAD

**Issue by spunkedy** [juju-solutions/bundle-canonical-kubernetes#272]
Basically this similar stack overflow.

http://stackoverflow.com/questions/41902127/kubernetes-getting-internal-error-occurred-failed-to-allocate-a-serviceip-rang

having something besides a /24 as a default might be good

**Comment by chuckbutler** [comment]
I think we're good to swap our default with a /16. I'll shop this around the team and get some testing in on this configuration to ensure it doesn't cause any issues. I don't expect that it will but I'm leaning more towards test first release later than breaking our defaults.

Thanks for reporting this @spunkedy

**Comment by chuckbutler** [comment]
@spunkedy I followed up with a co-worker of yours and with a redeploy the extension of the subnet to a /16 seems to have resolved the immediate issue. 

@wwwtyro @Cynerva @marcoceppi - How do we feel about making the subnet larger by default? This brings a slightly larger risk of IP collision in enterprise environments, and since it's an immutable configuration option - I'm a bit concerned about blindly increasing it without some supporting votes to this, however, it's a one line change.

**Comment by marcoceppi** [comment]
What would it take to make this mutable?

**Comment by Cynerva** [comment]
> What would it take to make this mutable?

We'll have to delete and recreate every Service object in the Kubernetes cluster. And since the IP of the kube-dns service will change, we'll have to reconfigure kubelet and recreate all the pods too. Fun stuff!

**Comment by spunkedy** [comment]
Personal opinion. I think having the /16 by default is fine. 

Making the option mutable would be nice for reasons unrelated to this issue though. If you wanted to change the cidr post deployment for some reason of network collision, then it might be nice.

**Comment by chuckbutler** [comment]
I agree with you @spunkedy. I think the comment by @Cynerva  was that it's not a trivial change and reconfiguring the application will effectively bring down the entire cluster at once in order to reconfigure.

**Comment by chuckbutler** [comment]
Looks to me like we still need to update this in the master charm

https://github.com/kubernetes/kubernetes/blob/master/cluster/juju/layers/kubernetes-master/config.yaml#L10-L12

**Comment by Cynerva** [comment]
In testing, we found that changing the default service-cidr causes a problem in our upgrade path.

After upgrading the master, Juju provides the new 10.152.0.0/16 value to the charm, which triggers the `WARN: cannot change service-cidr, still using 10.152.183.0/24` status message.

This isn't too bad, since the admin can simply set the config back to 10.152.183.0/24 manually to clear the message.

However, there's a bigger concern. When the service-cidr changes, existing kubernetes-master units will continue to work fine, since they stick to the old value stored in the unit DB. But if any new units of kubernetes-master are created during this time, they will stupidly use the new value and potentially cause serious problems for the cluster.

For now, we are reverting this change via https://github.com/kubernetes/kubernetes/pull/51200 until we can revisit this with a safer fix.
